File: app/src/main/java/com/example/colorfulcircle/MainActivity.java
package com.example.colorfulcircle;

import android.graphics.drawable.GradientDrawable;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import com.example.colorfulcircle.databinding.ActivityMainBinding;
import java.util.Random;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    private GradientDrawable generateRandomColorDrawable() {
        Random random = new Random();
        int color = 0xff000000 | random.nextInt(0xffffff + 1);  // Add the alpha channel
        GradientDrawable drawable = new GradientDrawable();
        drawable.setShape(GradientDrawable.OVAL);
        drawable.setColor(color);
        return drawable;
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());

        binding.circle.setOnClickListener(v -> binding.circle.setBackground(generateRandomColorDrawable()));

        setContentView(binding.getRoot());
    }
}

File: app/src/main/java/com/example/colorfulcircle/MainActivity.kt
package com.example.colorfulcircle

import android.annotation.SuppressLint
import android.graphics.Color
import android.graphics.drawable.GradientDrawable
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.colorfulcircle.databinding.ActivityMainBinding
import kotlin.random.Random

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private fun generateRandomColorDrawable(): GradientDrawable {
        val random = Random
        val color = String.format("#%06x", random.nextInt(0xffffff + 1))
        val drawable = GradientDrawable()
        drawable.shape = GradientDrawable.OVAL
        drawable.setColor(Color.parseColor(color))
        return drawable
    }

    @SuppressLint("UseCompatLoadingForDrawables")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)

        binding.apply {
            circle.setOnClickListener {
                circle.background = generateRandomColorDrawable()
            }
        }

        setContentView(binding.root)

    }
}